### DISCUSSION TITLE
Source Maps in autoBuild?

### DISCUSSION THREAD

**Discussion by jrochkind** [ElMassimo/vite_ruby#285] [Q&A]
In actual production, I seem to be getting sourceMaps out of vite with `rake assets:precompile`

But when using `autoBuild` feature in development, I don't seem to be getting sourceMaps. This is a bit inconvenient when in dev or test using autoBuild, the point when I most need sourceMaps actually! 

Is there any way to get sourceMaps when under `autoBuild`?  The [documentation](https://vite-ruby.netlify.app/config/#source-maps-%F0%9F%97%BA) suggests that Source Maps should already be on by default, and has instructions for turning them off... 

Ah, but wait, adding `build: { sourcemap: true }` _does_ seem to have gotten me source maps in `autoBuild` mode!  

If I wanted to, how would I enable them in dev autoBuild but _disable_ them in production?

**Comment by ElMassimo** [comment]
> If I wanted to, how would I enable them in dev autoBuild but disable them in production?

For example:

```ts
build: { sourcemap: process.env.VITE_RUBY_AUTO_BUILD === "true"  }
```

**Reply by jrochkind** [reply]
Ah, thanks!

**Reply by jrochkind** [reply]
I guess maybe this would also work?

    build: { sourcemap: (process.env.NODE_ENV != "production")  }

**Reply by ElMassimo** [reply]
Yeah, depending on your deployment configuration, that should work as well.

**Reply by jrochkind** [reply]
For anyone reading along....

`process.env.VITE_RUBY_AUTO_BUILD` is actually a string value, `"true"`, or `"false"`.  

The Javascript string `"false"` actually IS truthy in JS, so it looks to me you may actually need to do something like:

    build: { sourcemap: (process.env.VITE_RUBY_AUTO_BUILD == "true) }

NOT just `sourcemap: process.env.VITE_RUBY_AUTO_BUILD` as @ElMassimo suggested above!

Or maybe the string `"false"` vs `"true"` will actually work to do what is needed above? I was in a slightly different circumstance, where I actually was using a JS conditional with `process.env.VITE_RUBY_AUTO_BUILD`, and had a really confusing situation where it wasn't doing what I expected... until I realized it was `"true"` or `"false"`, and `if (process.env.VITE_RUBY_AUTO_BUILD)` was always true!

**Reply by ElMassimo** [reply]
Updated the answer to cast the value using `Boolean`.

**Reply by jrochkind** [reply]
Thanks! I am not very wise in JS, I didn't know `Boolean()`. 

I'm actually using the check not for sourcemaps, but to avoid creating `.gz` alternates in autoBuild mode -- it slows down autoBuild significantly, and .gz variant isn't really needed in development, although I want it in production so the rails app will see it and serve it directly (as used on heroku).

**Reply by jrochkind** [reply]
@ElMassimo Nope, `Boolean()` is no good here. 

`process.env.VITE_RUBY_AUTO_BUILD` can be string `"true"` or `"false"`. 

`Boolean("false")` is `true` though. No good. 

I'm going with `process.env.VITE_RUBY_AUTO_BUILD == "true"`

Please don't change `process.env.VITE_RUBY_AUTO_BUILD` from a string to a boolean though, as it'll now break me!

**Reply by ElMassimo** [reply]
Makes sense, I'm using `Boolean` to ignore empty strings, but in this case it will pass `false` explicitly.

Env variables are always strings.
